tests/tests.cc

```
@@ -780,6 +781,713 @@ class Tests {
+      minio::s3::FileHeaderInfo file_header_info =
+          minio::s3::FileHeaderInfo::kUse;
+      csv_input.file_header_info = &file_header_info;
+      minio::s3::CsvOutputSerialization csv_output;
+      minio::s3::QuoteFields quote_fields = minio::s3::QuoteFields::kAsNeeded;
+      csv_output.quote_fields = &quote_fields;
+
+      // SelectRequest and serialization objects live in caller's stack.
+      // SelectObjectContentAsync must own copies to avoid dangling.
+      minio::s3::SelectRequest request(expression, &csv_input, &csv_output);
+      std::string records;
+      auto func = [&records](minio::s3::SelectResult result) -> bool {
```

Cppcheck (2.21.0)

[performance] Function parameter 'result' should be passed by const reference.

(passedByValue)